src/components/Shifting/BoardView.tsx

```diff
@@ -35,16 +36,16 @@ export default function BoardView() {
   });
 
   const [modalFor, setModalFor] = useState<{
-    externalId?: string;
+    external_id: string | undefined;
```

Comment:
Can keep this as `external_id?: string`.

LGTM.

Reply:
Added